website/docs/docs/build/unit-tests.md

```diff
@@ -0,0 +1,253 @@
+---
+title: "Unit tests"
+sidebar_label: "Unit tests"
+description: "Read this tutorial to learn how to use unit tests on your SQL models."
+search_weight: "heavy"
+id: "unit-tests"
+keywords:
+  - unit test, unit tests, unit testing, dag
+---
+:::note closed beta
+
+Unit testing is currently in closed beta for dbt Cloud accounts that have updated to a [versionless environment](/docs/dbt-versions/upgrade-core-in-cloud).
+
+It is available now as an alpha feature for dbt Core v1.8 users.
+
+:::
+
+Historically, the test coverage capabilities of dbt were limited to [“data” tests](/docs/build/data-tests) &dash; testing the quality of input data or the shape of the resulting datasets &mdash; that could only be executed _after_ a model had been built. 
+
+Now, we are introducing a new type of test to dbt - unit tests. In software programming, unit tests validate small portions of your functional code, and they work much the same way here. Unit tests allow you to validate your SQL modeling logic on a small set of static inputs _before_ you materialize your full model in production. Unit tests enable test-driven development, benefiting developer efficiency and code reliability. 
+
+:::note Before you begin
+
+- We currently only support unit testing SQL models.
+- We currently only support adding unit tests to models in your _current_ project.
+- If your model has multiple versions, by default the unit test will run on *all* versions of your model. Read [unit testing versioned models for more information](#unit-testing-versioned-models).
+
+Read the [reference doc](/reference/resource-properties/unit-tests) for more details about formatting your unit tests.
+:::
+
+Let’s say you’re creating a new `dim_customers` model with a field `is_valid_email_address`, that calculates whether or not the customer’s email is valid: 
+
+<file name='dim_customers.sql'>
+
+```sql
+with customers as (
+
+    select * from {{ ref('stg_customers') }}
+
+),
+
+accepted_email_domains as (
+
+    select * from {{ ref('top_level_email_domains') }}
+
+),
+	
+check_valid_emails as (
+
+    select
+        customers.customer_id,
+        customers.first_name,
+        customers.last_name,
+	coalesce (regexp_like(
+            customers.email, '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$'
+        )
+        = true
+        and accepted_email_domains.tld is not null,
+        false) as is_valid_email_address
+    from customers
+		left join accepted_email_domains
+        on customers.email_top_level_domain = lower(accepted_email_domains.tld)
+
+)
+
+select * from check_valid_emails
+```
+</file>
+
+This type of logic can be challenging to validate. You can add a unit test to this model to ensure your `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
```

Comment:
```suggestion
The logic posed in the previous example can be challenging to validate. You can add a unit test to this model to ensure the `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
```

Reply:
This, to me, reads like we're referencing the example that came before the one we just finished going through. How about
```suggestion
The logic posed in this example can be challenging to validate. You can add a unit test to this model to ensure the `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
```